subprojects/launcher/launcher.gradle

```diff
@@ -90,7 +94,7 @@ class StartScriptGenerator extends DefaultTask {
         generator.exitEnvironmentVar = 'GRADLE_EXIT_CONSOLE'
         generator.mainClassName = 'org.gradle.launcher.GradleMain'
         generator.scriptRelPath = 'bin/gradle'
-        generator.classpath = ["lib/${launcherJar.singleFile.name}" as String]
+        generator.classpath = ["lib/${launcherJar.singleFile.name}" as String, "lib/asm-all-${asmVersion}.jar" as String]
```

Comment:
Am I assuming correctly that this classpath is only used at a later point: https://github.com/gradle/gradle/blob/3573b842c9a61a2bb4c5b532ab961fabbcf67cb4/subprojects/launcher/launcher.gradle#L48-L48 ?